src/solvers/Lanczos.jl

```diff
@@ -255,6 +255,10 @@ function (c::StopWhenFirstOrderProgress)(
     dmp::AbstractManoptProblem{<:TangentSpace}, ls::LanczosState, i::Int
 )
     if (i == 0)
+        if norm(ls.X) == 0
```

Comment:
```suggestion
        if norm(ls.X) == zero(eltype(ls.X))
```

Do you expect an exact zero btw, this might be more robust with an \approx 0 check

Reply:
Do you expect an exact zero btw, this might be more robust with an \approx 0 check?

Reply:
Fo now I mean exact zero. We can also introduce a tolerance, but the two cases where that appeared (and I know for both why) it was _exact_ zero.

1. Mateusz example started at a minimiser and the vector was literally `[0.0, 0.0]`.
2. your example had _just_ maxiter as stopping criterion, so it would have continued 10k iterations if you set that. At iteration 17 it ran below numerical precision and hence in tier 18 the norm of the gradient was again exactly 0.

Reply:
The first I would consider a bug,
the second maybe half-a-bug; really just maxiter as a stopping criterion is not the best of choices.

Reply:
Even when now trying to write a test, I just managed to get a first gradient of norm `1e-15` and that did not result in `NaN`s. So one really needs an example with _exactly_ zero gradient for Lanczos to fail.

Reply:
> this might be more robust with an \approx 0 check?

This is actually a common misconception: for Float64 `x == 0` and `isapprox(0, x)` are equivalent, see:
```julia
julia> isapprox(0.0, nextfloat(0.0))
false
```
Maybe it would be better to have some positive tolerance but it's not obvious without looking carefully at the algorithm.

Reply:
I can add more comments, but I was even with an hour of coming up with examples not able to find a gradient that was not _exactly_ zero and were Lanczos failed.
For example the new Test for ARC the gradient at the (numeric/from Eigenvectors) minimiser has only entries of size `1e-16` its norm is of size `1e-15`,
but still Lanczos runs fine and the special case introduced here is not needed (I mean it still stops after one step, but Lanczos is fine with not producing NaNs). So I feel _exact zero_ is the only problem here.